xbmc/cores/dvdplayer/DVDPlayer.cpp

```diff
@@ -835,7 +835,7 @@ bool CDVDPlayer::OpenDemuxStream()
   int64_t len = m_pInputStream->GetLength();
   int64_t tim = m_pDemuxer->GetStreamLength();
   if(len > 0 && tim > 0)
-    m_pInputStream->SetReadRate((unsigned int) (g_advancedSettings.m_readBufferFactor * len * 1000 / tim));
```

Comment:
When technically it's not wrong but it has an awkward side-effect. If a user increases the bufferfactor here, we complain (using an toast dialog) when the readrate*factor is too low compared to the average-rate of the file being played. Instead we only want to see this toast-dialog if the actual readrate < averagerate.